aws/lambda/oss_ci_job_queue_time/lambda_function.py

```diff
@@ -0,0 +1,699 @@
+#!/usr/bin/env python
+import argparse
+import io
+import json
+import logging
+import os
+import gzip
+import threading
+import yaml
+
+import boto3  # type: ignore[import]
+import clickhouse_connect
+from datetime import datetime, time
+
+# Local imports
+from functools import lru_cache
+from logging import info
+from typing import Any, Optional, Dict, Set, Iterable, List, Tuple
+from github import Github, Auth
+from dateutil.parser import parse
+
+
+logging.basicConfig(level=logging.INFO)
+_bucket_name = "ossci-raw-job-status"
+
+_in_queue_job_select_statement = """
+SELECT
+    DATE_DIFF(
+        'second',
+        job.created_at,
+        {timestamp:DateTime}
+    ) AS queue_s,
+    workflow.repository.'full_name' AS repo,
+    workflow.name AS workflow_name,
+    job.name AS job_name,
+    job.html_url,
+    IF(
+        LENGTH(job.labels) = 0,
+        'N/A',
+        IF(
+            LENGTH(job.labels) > 1,
+            job.labels[2],
+            job.labels[1]
+        )
+    ) AS machine_type,
+    toUnixTimestamp({timestamp:DateTime}) AS time
+FROM
+    default.workflow_job job FINAL
+    JOIN default.workflow_run workflow FINAL ON workflow.id = job.run_id
+"""
+
+
+@lru_cache()
+def get_clickhouse_client(host: str, user: str, password: str) -> Any:
+    # for local testing only, disable SSL verification
+    # return clickhouse_connect.get_client(host=host, user=user, password=password, secure=True, verify=False)
+
+    return clickhouse_connect.get_client(
+        host=host, user=user, password=password, secure=True
+    )
+
+
+@lru_cache()
+def get_aws_s3_resource() -> Any:
+    return boto3.resource("s3")
+
+
+def get_clickhouse_client_environment() -> Any:
+    for env in ["CLICKHOUSE_ENDPOINT", "CLICKHOUSE_USERNAME", "CLICKHOUSE_PASSWORD"]:
```

Comment:
Don't use environment variables directly, and any parameter that your function requires should come as parameters.